src/index.ts

```diff
@@ -13,49 +13,94 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import * as extend from 'extend';
 import * as path from 'path';
-import {AuthenticationConfig, Common, ServiceConfig} from '../third_party/types/common-types';
+import * as request from 'request';
+
+import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
+
 import {Config, defaultConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
+const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
+
+// Returns a promise which will resolve to true if running on GCE and false
+// otherwise.
+// It is assumed a program is running on GCE if the metadata API can be
+// accessed.
+function onGCE(): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
+    request.get(
+        {url: metadataAPI, headers: {'Metadata-Flavor': 'Google'}},
+        function(err: Error, response) {
+          err ? resolve(false) : resolve(true);
+        });
+  });
+}
+
+// Returns a promise that resolves to the value of metadata field.
+// Promise will be rejected if there is a problem accessing metadata API.
+function getField(field: string): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    request.get(
+        {url: metadataAPI + field, headers: {'Metadata-Flavor': 'Google'}},
+        function(err: Error, response: request.RequestResponse) {
+          err ? reject(err) : resolve(response.body);
+        });
+  });
+}
 
 // initConfig sets unset values in the configuration to the value retrieved from
 // environment variables, metadata, or the default value specified in
 // defaultConfig.
 // Returns rejected promise if value that must be set cannot be initialized.
-async function initConfig(config: Config): Promise<ProfilerConfig> {
+// Exported for testing purposes.
+export async function initConfig(config: Config): Promise<ProfilerConfig> {
   config = common.util.normalizeArguments(null, config);
 
-  const envConfig = {
-    logLevel: process.env.GCLOUD_PROFILER_LOGLEVEL,
+  const envConfig: any = {
     projectId: process.env.GCLOUD_PROJECT,
     serviceContext: {
       service: process.env.GAE_SERVICE,
       version: process.env.GAE_VERSION,
     }
   };
 
+  if (process.env.GCLOUD_PROFILER_LOGLEVEL !== undefined) {
```

Comment:
Do I read it correct that if I set GCLOUD_PROFILER_LOGLEVEL to something that is not a number (like "foo"), the logging will be enabled at a rather verbose level 10?

Reply:
The second argument means parseInt() will parse values to be in base 10.
If GCLOUD_PROFILER_LOGLEVEL is not set, or set to something that can't be parsed as an int, it will be ignored.